src/pages/ChangePassword.tsx

```diff
@@ -13,11 +13,26 @@ export default function ChangePassword(): JSX.Element {
 
     const server = useSelector((state: apiReducerState) => state.server);
     const [searchParams, setSearchParams] = useSearchParams();
+    
+    // parseJwt is currently untested
+    function parseJwt (token: any) {
+        if (token === null) {
+            return "";
+        }
+        var base64Url = token.split('.')[1];
+        var base64 = base64Url.replace(/-/g, '+').replace(/_/g, '/');
+        var jsonPayload = decodeURIComponent(window.atob(base64).split('').map(function(c) {
+            return '%' + ('00' + c.charCodeAt(0).toString(16)).slice(-2);
+        }).join(''));
+    
+        return JSON.parse(jsonPayload);
+    };
 
     async function ChangePass() {
         // Check newPass and conPass to be the same and alert if not
         const apikey = searchParams.get("token")
-        console.log(apikey)
+        username = parseJwt(apikey)
```

Comment:
The username is part of the object, not the whole object. Need to create an interface and type this.